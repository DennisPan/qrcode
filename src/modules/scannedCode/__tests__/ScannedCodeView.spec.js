
/* eslint-disable no-undef */
import React from 'react';
import Enzyme from 'enzyme';

import {
  ShareButton,
  CopyButton,
  OpenButton,
  capitalizeFirstLetter,
} from '../ScannedCodeView';

describe('ScannedCode => ShareButton', () => {
  it('renders as expected', () => {
    const wrapper = Enzyme.shallow(
      <ShareButton
        data="https://insiderdev.com"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ScannedCode => CopyButton', () => {
  it('renders as expected', () => {
    const wrapper = Enzyme.shallow(
      <CopyButton
        data="https://insiderdev.com"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ScannedCode => OpenButton', () => {
  it('renders as expected', () => {
    const wrapper = Enzyme.shallow(
      <OpenButton
        data="https://insiderdev.com"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('capitalizeFirstLetter', () => {
  it('should work properly', () => {
    expect(capitalizeFirstLetter('hello world')).toBe('Hello world');
    expect(capitalizeFirstLetter('Hello world')).toBe('Hello world');
    expect(capitalizeFirstLetter('')).toBe('');
  });
});
