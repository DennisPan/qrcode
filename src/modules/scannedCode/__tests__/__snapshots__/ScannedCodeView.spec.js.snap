// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScannedCode => CopyButton renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CopyButton
    data="https://insiderdev.com"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "copy",
      "onPress": [Function],
      "variant": "primary",
    },
    "ref": null,
    "rendered": "copy",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "copy",
        "onPress": [Function],
        "variant": "primary",
      },
      "ref": null,
      "rendered": "copy",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ScannedCode => OpenButton renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpenButton
    data="https://insiderdev.com"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "open",
      "onPress": [Function],
      "variant": "primary",
    },
    "ref": null,
    "rendered": "open",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "open",
        "onPress": [Function],
        "variant": "primary",
      },
      "ref": null,
      "rendered": "open",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ScannedCode => ShareButton renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShareButton
    data="https://insiderdev.com"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "share",
      "onPress": [Function],
      "variant": "primary",
    },
    "ref": null,
    "rendered": "share",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "share",
        "onPress": [Function],
        "variant": "primary",
      },
      "ref": null,
      "rendered": "share",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
